apps/aecore/test/aec_chain_sim.erl

```diff
@@ -407,28 +429,31 @@ fork_switch_(ForkId, #{forks := Forks, mempool := Pool, orphans := Orphans} = Ch
     NewChain = Chain#{ forks => NewForks
                      , mempool => NewPool
                      , orphans => Evict ++ Orphans },
-    {ok, announce(main, [], NewChain)}.
+    {ok, announce(main, [], NewChain, Opts)}.
 
 %% Announce top_changed and tx events
-announce(ForkId, Txs, #{ forks := Forks } = Chain) ->
+announce(ForkId, Txs, #{ forks := Forks } = Chain, Opts) ->
     #{ ForkId := #{ blocks := [#{ hash := TopHash
                                 , prev := PrevHash
                                 , header := Hdr } | _] = Blocks} } = Forks,
+    Simulator = simulator(Opts),
     Height = length(Blocks) + 1,
     Type = aec_headers:type(Hdr),
     Origin = origin(Type),
-    Info = #{ block_hash   => TopHash
+    Info = #{ pid => self(),
+              block_hash   => TopHash
             , block_type   => Type
             , block_origin => Origin
             , prev_hash    => PrevHash
             , height       => Height },
-    send_tx_events(Txs, Info),
-    case ForkId of
-        main ->
-            ?LOG("Publishing top_changed, I = ~p", [Info]),
-            aec_events:publish(top_changed, Info);
-        _ ->
-            ignore
+    if Simulator == parent_chain ->
+        aec_events:publish({parent_chain, top_changed}, Info#{txs => Txs});
+        true ->
+            send_tx_events(Txs, Info),
+            (ForkId == main) andalso
```

Comment:
I don't like this folding of the case statement

Reply:
May do some decoration after (I got your style notices in general, see no problem here)